crates/client/src/lib.rs

```diff
@@ -101,6 +102,36 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
         &self.namespace_map
     }
 
+    /// Get
+    pub async fn get_warg_registry(&self, namespace: &str) -> Result<Option<RegistryDomain>> {
+        self.update_checkpoint(None, &self.api.latest_checkpoint(None).await?, vec![])
+            .await?;
+        let operator = self
+            .registry()
+            .load_operator(Some(&RegistryDomain::from_str(namespace)?))
+            .await?;
+        if let Some(op) = operator {
+            let namespace_state = op.state.namespace_state(namespace);
+            if let Ok(Some(warg_protocol::operator::NamespaceState::Imported { registry })) =
+                namespace_state
+            {
+                return Ok(Some(RegistryDomain::from_str(registry)?));
+            } else if let Ok(Some(warg_protocol::operator::NamespaceState::Defined)) =
+                namespace_state
+            {
+                return Ok(None);
+            };
```

Comment:
nit: You'd typically use `match` here:

```suggestion
            match op.state.namespace_state(namespace) {
                Ok(Some(warg_protocol::operator::NamespaceState::Imported { registry })) => {
                    return Ok(Some(RegistryDomain::from_str(registry)?));
                }
                Ok(Some(warg_protocol::operator::NamespaceState::Defined)) => {
                    return Ok(None);
                }
                _ => (),
            }
```